### DISCUSSION TITLE
[SDK] Check ScanDevice capabilities mainly feeder/duplex support.

### DISCUSSION THREAD

**Discussion by SamehAmnoun** [cyanfish/naps2#400] [Development]
Hello, 

I am using the NAPS2.SDK Nuget package, and I want to check if a certain scan device supports feeder/duplex scanning without actually attempting to scan. Similar to device.SupportsFeeder() method which is available in NAPS.Wia. So, is there a way to achieve this that I am missing?

**Comment by AOgai** [comment]
Guten Tag,

there is a [GetCaps](https://www.naps2.com/sdk/doc/api/NAPS2.Scan.ScanController.html?q=GetCaps#NAPS2_Scan_ScanController_GetCaps_NAPS2_Scan_ScanDevice_CancellationToken_) method for the ScanController. (#183)

Example code:

```
using (var scanningContext = new ScanningContext(new WpfImageContext()))
{
    var controller = new ScanController(scanningContext);
    ScanDevice device = (await controller.GetDeviceList(Driver.Wia)).First();
    var caps = await controller.GetCaps(device);
};
```

:-)
Arnold
